src/Compilers/CSharp/Portable/Generated/Syntax.xml.Internal.Generated.cs

```diff
@@ -12284,14 +12284,20 @@ static LocalFunctionStatementSyntax()
 
   internal sealed partial class LocalDeclarationStatementSyntax : StatementSyntax
   {
+    internal readonly SyntaxToken usingKeyword;
```

Comment:
i'm not s huge fan of repurposing this node in this manner.  It feels more appropriate as a new sort of node.
But i could be convinced otherwise.

Reply:
I rather like it. Tagging @gafter for advice on the shape of the syntax tree (extend local declarations, or create a dedicated kind of node?).

---
In reply to: [200508613](https://github.com/dotnet/roslyn/pull/28315#discussion_r200508613) [](ancestors = 200508613)

Reply:
This is usually what we do when we augment nodes. For instance, when we added `ref` locals we didn't add new types of local declarations, we just added optional `ref` modifiers. I think it's the same situation here.

Reply:
Yeah, i'm on the fence.  Something feels... 'wonky' here.  Perhaps because 'using' bridges between two things (using-statements, and local-var-statements).  That's unlike 'ref'.

That said, this may just need some time on my part to get used to it.  So this is not at all strong pushback.  If the team feels like this is "good", then i have no objection.

Reply:
Could it be parsed as the modifier (LocalDeclarationStatementSyntax.Modifiers)?

Reply:
Could it be parsed into `Modifiers` on LocalDeclaration? Currently Modifiers is only used for `const`.